simulator/risc_v/riscv_register/riscv_register.h

```diff
@@ -45,10 +52,10 @@ class RISCVRegister {
     bool is_zero()                 const { return value == RISCV_REG_zero; }
     constexpr bool is_mips_hi()    const { return false; }
     constexpr bool is_mips_lo()    const { return false; }
-    static auto from_cpu_index( uint8 id) { return RISCVRegister( RegNum{ id}); }
-    static auto from_gdb_index( uint8 id) { return RISCVRegister( RegNum{ id}); }
-    static auto from_csr_index( uint16 id) { return RISCVRegister( get_csr_regnum( id)); }
-    static auto from_csr_name( std::string_view name) { return RISCVRegister( get_csr_regnum( name)); }
+    static RISCVRegister from_cpu_index( uint8 id = uint8( MAX_REG), separator type = separator::ordinary);
+    static RISCVRegister from_gdb_index( uint8 id = uint8( MAX_REG), separator type = separator::ordinary);
+    static RISCVRegister from_csr_index( uint16 id) { return RISCVRegister( get_csr_regnum( id)); }
```

Comment:
Please put `auto` back. If we decide to rename the class, we will have less type mismatch problems.